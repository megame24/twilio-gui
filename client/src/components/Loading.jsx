import React from 'react';
import { PropTypes } from 'prop-types';
import { Dimmer, Loader } from 'semantic-ui-react';

const Loading = ({ isLoading }) => (
  <Dimmer
    page
    active={isLoading}
  >
    <Loader size="medium" />
  </Dimmer>
);

Loading.propTypes = {
  isLoading: PropTypes.bool.isRequired,
};

export default Loading;
