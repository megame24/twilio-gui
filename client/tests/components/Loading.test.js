import React from 'react';
import { shallow } from "enzyme";
import Loading from '../../src/components/Loading';

describe('Testing Loading component', () => {
  it('Should render as expected', () => {
    const props = {
      isLoading: true
    }
    const tree = shallow(<Loading { ...props } />);
    expect(tree).toMatchSnapshot();
  });
});
