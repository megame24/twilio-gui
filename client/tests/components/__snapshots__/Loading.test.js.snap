// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Loading component Should render as expected 1`] = `
<Dimmer
  active={true}
  page={true}
>
  <Loader
    size="medium"
  />
</Dimmer>
`;
